packages/vite-node/src/source-map-handler.ts

```diff
@@ -192,8 +192,10 @@ function mapSourcePosition(position: OriginalMapping) {
     // better to give a precise location in the compiled file than a vague
     // location in the original file.
     if (originalPosition.source !== null) {
+      // `originalPositionFor` do not return the expected file source. So use `position.source`
+      // instead of `originalPosition.source`
       originalPosition.source = supportRelativeURL(
-        sourceMap.url, originalPosition.source)
+        sourceMap.url, position.source)
```

Comment:
Doesn't look like this is the fix. `originalPositionFor` should return the correct position. Is something wrong with the map?

Reply:
1. function `originalPositionFor` works well, but the arguemnt `sourceMap.map` is broken, thus `originalPositionFor ` doesn't return the correct position. It seems that `ViteDevServer` doesn't return the `sources` in absolute path (`/home/cattchen/codes/github.com/DanHulton/vitest-stack-sourcemap-repro/src/main.js` is expected, but finally got `/src/main.js`). I suppose that it's better to use `position.source` than `originalPosition.source`

![image](https://user-images.githubusercontent.com/42082890/230020377-7a6c3b60-7ea4-43d8-bd45-6a7ef1f8e235.png)

![image](https://user-images.githubusercontent.com/42082890/230021754-41e8b9e8-e7df-42d7-9905-5b8f49263b29.png)

2. `sourceMap.url` and `supportRelativeURL` seems work well.

Reply:
Let's patch map's `sources` in `packages/vite-node/src/server.ts:195` then. I think `toFilePath` should return the correct path.

Reply:
Thanks for your suggestions! It seems to work well.

![image](https://user-images.githubusercontent.com/42082890/230077661-5e03a88b-8060-42dc-8ce8-f375eebdb3a9.png)

![image](https://user-images.githubusercontent.com/42082890/230078044-633713e0-2e61-41ea-b261-44ea3dab4c95.png)